### PR TITLE
OSIDB-3004: Add suggestions to flaw owner field

### PR DESCRIPTION
**Pull request by MrMarble**
# OSIDB-3004: Add suggestions to flaw owner field
## Checklist:

- [x] Linting passed
- [x] Type checks passed
- [x] Tests suite passed
- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

Added a dropdown suggestion integrated with jira

image

Closes OSIDB-3004

### CODE REVIEW

File: src/components/widgets/EditableTextWithSuggestions.vue [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        >{{ modelValue === '' ? placeholder : modelValue }}</span>
+        <!--if a button is inside a label, clicking the label clicks the button?-->
+        <button
+          v-if="!readOnly"
+          type="button"
+          class="osim-editable-text-pen input-group-text"
+          tabindex="-1"
+          @click="beginEdit"
+        ><i class="bi bi-pencil"></i></button>
+        <slot name="buttons-out-of-editing-mode" v-bind="{ commit, onBlur }"></slot>
```
Comment (C-Valen) [inline src/components/widgets/EditableTextWithSuggestions.vue:106]: I don't understand why we do require two different slots `buttons-out-of-editing-mode` and `buttons-in-editing-mode`

**Review by C-Valen — COMMENTED** [review]
Small detail: 
- Self assign button border radius change when the input is focused

Possible UI/UX improvements:
- Add some feedback to the user always that a fetch is being performed (e.g. hiding previous results and showing a loading spinner)
- Limit suggestion box width to the field's width, currently the width is based on the longest suggested item (and may be truncating very long full names). This would be less relevant after you rebase over the new wider form style, but I think is still recommendable for UI consistency.
- For some users there are two options, I think we could filter those cases to use only the one that has an email associated

File: src/components/widgets/EditableTextWithSuggestions.vue (hunk 1 above)
Comment (MrMarble) [reply src/components/widgets/EditableTextWithSuggestions.vue:106]: It was like that, is part of the `EditableText` widget, I copied the code to add the suggestions part so the API is the same on both

**Comment by MrMarble** [comment]
> For some users there are two options, I think we could filter those cases to use only the one that has an email associated

I think that is only on stage, in production there's only 1 user, also we cannot filter, because not all users have the email as the user id (mine for example)

File: src/components/widgets/DropDown.vue [hunk 2]
```diff
@@ -0,0 +1,42 @@
+<script setup lang="ts">
+defineProps<{
+  show: boolean,
+}>();
```
[the remark this answers is not in the document]
Comment (MrMarble) [reply src/components/widgets/DropDown.vue:4]: Yeah that is easier 😆

**Review by C-Valen — APPROVED** [review]

**Review by tboyetteredhat — APPROVED** [review]

**Review by superbuggy — APPROVED** [review]

**Pull request merged**